### PR TITLE
add HttpFetcher

### PR DESCRIPTION
**Pull request by fspot**
Add support for scheme http/https

### CODE REVIEW

File: peakina/datasource.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class DataSource:
     uri: str
     type: TypeEnum = None
     match: MatchEnum = None
+    scheme: str = None
```
Comment (PrettyWood) [inline peakina/datasource.py]: Why adding `scheme` here since it's already in `__post_init__` ? IDE warning?

**Review by PrettyWood — CHANGES_REQUESTED** [review]
Except a small remark LGTM !

File: peakina/datasource.py (hunk 1 above)
Comment (fspot) [reply peakina/datasource.py]: tests were failing https://circleci.com/gh/ToucanToco/peakina/77?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link, it looks like `__post_init__` is not allowed to write new attributes :thinking:

Comment (PrettyWood) [reply peakina/datasource.py]: Yet tests work fine on my machine and used to work before 🤔

File: peakina/io/http/http_fetcher.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        r = PoolManager().request('GET', filepath, preload_content=False)
+        ret = tempfile.NamedTemporaryFile(suffix='.httptmp')
+        for chunk in r.stream():
+            ret.write(chunk)
+        ret.seek(0)
+        return ret
+
+    @staticmethod
+    def listdir(dirpath) -> List[str]:
+        raise NotImplementedError
```
Comment (PrettyWood) [inline peakina/io/http/http_fetcher.py]: Could you add a test for this please?

**Review by PrettyWood — APPROVED** [review]

**Pull request merged**